src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -155,6 +240,212 @@ def truncate_table(self, table):
         """Truncate table"""
         self.run_sql(f"TRUNCATE {self.get_table_qualified_name(table)}")
 
+    def is_native_autodetect_schema_available(self, file: File) -> bool:
+        """
+        Check if native auto-detection of schema is available.
+
+        :param file: File used to check the file type of to decide whether there is a native auto-detection available.
+        """
+        supported_config = self.NATIVE_AUTODETECT_SCHEMA_CONFIG.get(file.location.location_type)
+        if supported_config and file.type.name in supported_config["filetype"]:  # type: ignore
+            return True
+        return False
+
+    def create_table_using_native_schema_autodetection(
+        self,
+        table: Table,
+        file: File,
+    ) -> None:
+        """
+        Create a SQL table, automatically inferring the schema using the given file via native database support.
+
+        :param table: The table to be created.
+        :param file: File used to infer the new table columns.
+        """
+        supported_config = self.NATIVE_AUTODETECT_SCHEMA_CONFIG.get(file.location.location_type)
+        if not supported_config:
+            raise ValueError("Unsupported file location.")
+        return supported_config["method"](table=table, file=file)  # type: ignore
+
+    # Require skipcq because method overriding we need param target_table
+    def is_native_path_available(  # skipcq: PYL-R0201
+        self,
+        source_dataset: File | Table,  # skipcq PYL-W0613
+    ) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
```

Comment:
It seems we don't have parm `target_table` here. Is this intended? Or should we remove it?

Reply:
[Done](https://github.com/astronomer/apache-airflow-provider-transfers/pull/66/commits/49dea8f8e568dc9b61f2dbb54d340c4c42f6ead1)